dbf-reader/src/main/java/org/jamel/dbf/structure/DbfRow.java

```diff
@@ -0,0 +1,158 @@
+package org.jamel.dbf.structure;
+
+import org.jamel.dbf.exception.DbfException;
+
+import java.math.BigDecimal;
+import java.nio.charset.Charset;
+import java.util.Date;
+
+import static org.jamel.dbf.utils.DbfUtils.trimLeftSpaces;
+
+public class DbfRow {
+
+    private DbfHeader header;
+    private Charset charset;
+    private Object[] row;
+
+    public DbfRow(DbfHeader header, Charset charset, Object[] row) {
+        this.header = header;
+        this.charset = charset;
+        this.row = row;
+    }
+
+    /**
+     * Retrieves the value of the designated field as java.math.BigDecimal.
+     *
+     * @param fieldName the name of the field
+     * @return the field value, or null (if the dbf value is NULL)
+     * @throws DbfException if there's no field with name fieldName
+     */
+    public BigDecimal getBigDecimal(String fieldName) throws DbfException {
+        Object value = get(fieldName);
+        return value == null ? null : new BigDecimal(value.toString());
+    }
+
+    /**
+     * Retrieves the value of the designated field as java.util.Date.
+     *
+     * @param fieldName the name of the field
+     * @return the field value, or null (if the dbf value is NULL)
+     * @throws DbfException if there's no field with name fieldName
+     */
+    public Date getDate(String fieldName) throws DbfException {
+        Date value = (Date) get(fieldName);
+        return value == null ? null : value;
+    }
+
+    /**
+     * Retrieves the value of the designated field as String.
+     *
+     * @param fieldName the name of the field
+     * @return the field value, or null (if the dbf value is NULL)
+     * @throws DbfException if there's no field with name fieldName
+     */
+    public String getString(String fieldName) throws DbfException {
```

Comment:
I think there also must be a method which allows to use different `charset` per column. In that case field `charset` must be renamed to `defaultCharset`.